includes/classes/Indexable/Post/SyncManager.php

```diff
@@ -107,6 +108,78 @@ public function action_queue_meta_sync( $meta_id, $object_id, $meta_key, $meta_v
 		}
 	}
 
+	/**
+	 * When a post's terms are changed, re-index
+	 *
+	 * This catches term deletions via wp_delete_term(), because that function internally loops over all attached objects
+	 * and updates their terms. It will also end up firing whenever set_object_terms is called, but the queue will de-duplicate
+	 * multiple instances per post
+	 *
+	 * @see set_object_terms
+	 * @param int    $object_id  Object ID.
+	 * @param array  $terms      An array of object terms.
+	 * @param array  $tt_ids     An array of term taxonomy IDs.
+	 * @param string $taxonomy   Taxonomy slug.
+	 * @param bool   $append     Whether to append new terms to the old terms.
+	 * @param array  $old_tt_ids Old array of term taxonomy IDs.
+	 * @since  3.5
+	 */
+	public function action_set_object_terms( $object_id, $terms, $tt_ids, $taxonomy, $append, $old_tt_ids ) {
+		$post_id = $object_id;
```

Comment:
Missing a check for `$this->kill_sync()`:

```suggestion
		if ( $this->kill_sync() ) {
			return;
		}
		$post_id = $object_id;
```